### PR TITLE
Fixed "Vehicle disappearing when grabbing another veh inside"

### LINKED ISSUE #34514: Vehicle disappearing when grabbing another veh inside(?)
# Describe the bug

When I try to grab lifter (one-tile vehicle with pallet lifter) which is in the middle of another vehicle being deconstructed, the latter disappears entirely!

# Steps To Reproduce

1. Load provided savegame.
2. Go out (south) of the evac, go to the right vehicle and grab away the lifter.
3. Notice how the under-construction vehicle disappears.

# Expected behavior

Veh not disappearing obviously.

# Screenshots

obraz

obraz

# Versions and configuration

- OS: Windows 8.1
- Game Version: 0.D-7950-ga718f4c [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Generic Guns [generic_guns],
    Alternative Map Key [alt_map_key]
]

# Additional context

Save game:
[Earlings.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3694610/Earlings.zip)

### PR DESCRIPTION
**Pull request by ipcyborg**
#### Summary
```SUMMARY: Bugfixes "Vehicle disappearing when grabbing another veh inside"```

#### Purpose of change
Fixes #34514
Function `map::displace_vehicle` takes only two coordinates, and tries to find the vehicle to displace by coordinate. 
This could lead to an issue when vehicle is detected incorrectly.
But it is not even required, since vehicle is already known. 

#### Describe the solution
The solution is to specify the vehicle in `map::displace_vehicle` parameters.

#### Describe alternatives you've considered
Probably some additional optimization, like removing `our_i`.

#### Testing
Tested on save file provided in #34514: only the lifter is grabbed, no vehicle disappeared.
Tested with `cata_test-tiles`.

#### Additional context
None

### CODE REVIEW

File: tests/vehicle_efficiency.cpp [hunk 1]
```diff
@@ -247,7 +247,7 @@ static int test_efficiency( const vproto_id &veh_id, int &expected_mass,
         // Bring it back to starting point to prevent it from leaving the map
         const tripoint displacement = starting_point - veh.global_pos3();
         tripoint veh_pos = veh.global_pos3();
-        g->m.displace_vehicle( veh_pos, displacement );
+        g->m.displace_vehicle( &veh, veh_pos, displacement);
```
Comment (mlangsdorf) [inline tests/vehicle_efficiency.cpp]:
```suggestion
        g->m.displace_vehicle( &veh, veh_pos, displacement );
```

File: src/map.cpp [hunk 2]
```diff
@@ -1007,7 +1007,7 @@ void map::unboard_vehicle( const tripoint &p, bool dead_passenger )
     unboard_vehicle( *vp, passenger, dead_passenger );
 }
 
-vehicle *map::displace_vehicle( tripoint &p, const tripoint &dp )
+bool map::displace_vehicle( vehicle *veh, tripoint &p, const tripoint &dp )
```
Comment (BevapDin) [inline src/map.cpp]: Would be better to accept a `vehicle` reference. Displacing a nullpointer does not make any sense and should not be allowed.

Comment (ipcyborg) [reply src/map.cpp]: Done, but I left other functions that uses vehicle pointer intact: like  `update_vehicle_cache` and `setremoteveh`. 
Changing them all could create a chain reaction. Should I do it anyway?

File: tests/vehicle_efficiency.cpp (hunk 1 above)
Comment (ipcyborg) [reply tests/vehicle_efficiency.cpp]: Will do, thanks!

**Pull request merged**